src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -1126,9 +1132,19 @@ export class EditDataGridPanel extends GridParentComponent {
 		});
 	}
 
-	onBeforeEditCell(event: Slick.OnBeforeEditCellEventArgs<any>): void {
+	onBeforeEditCell(event: Slick.OnBeforeEditCellEventArgs<any>): boolean {
+		let result = true;
 		this.logService.debug('onBeforeEditCell called with grid: ' + event.grid + ' row: ' + event.row
 			+ ' cell: ' + event.cell + ' item: ' + event.item + ' column: ' + event.column);
+
+		let itemToEdit = event.item[event.cell].displayValue;
+
+		if (Services.DBCellValue.isDBCellValue(itemToEdit) && itemToEdit.displayValue.indexOf('\u0000') !== -1) {
+			result = false;
+			this.notificationService.warn(cellWithNullMessage);
```

Comment:
Is this the best way to warn the users? I don't know if warning or error is more apt in this scenario

Reply:
Ideally this scenario never/rarely occurs, so I'm not sure a notification is too disruptive.  Ideally, showing some type of warning indicator directly on the cell would be better, but maybe overkill for rare corner-case.

Reply:
It is a warning as this is expected to happen and also given the current state of Edit Data, although yes an icon on the cell would be the best thing to do.